united-fare-finder.py

```diff
@@ -36,45 +36,96 @@
-elem.submit()
-
-time.sleep(10)
-
-# scrape fare data
-page_source = driver.page_source
-flight_elements = parsePageSource(page_source)
-
-# write to file
-for i in range(len(flight_elements)):
-    for j in range(len(flight_elements[i])):
-        print(*flight_elements[i][j])
-    print()
+origin = "ORD"
+dest = [
+  "hnl", "slc", "dfw", "lax", "sea", "atl",
+  "mia", "mco", "sfo", "jfk", "ewr", "lga", "dca", "iad"
+]
+depDate = "03/01/2023"
+retDate = "03/10/2023"
+
+count = 0
+for dest in dest:
+
+    # input flight info in main page
+    if (count == 0):
+        elem = driver.find_element(By.ID, "bookFlightOriginInput")
+        elem.clear()
+        elem.send_keys(origin)
+
+        elem = driver.find_element(By.ID, "bookFlightDestinationInput")
+        elem.clear()
+        elem.send_keys(dest)
+
+        elem = driver.find_element(By.ID, "DepartDate")
+        elem.clear()
+        elem.send_keys(depDate)
+
+        elem = driver.find_element(By.ID, "ReturnDate")
+        elem.clear()
+        elem.send_keys(retDate)
+
+        elem = driver.find_element(
+          By.XPATH, '//*[@id="bookFlightForm"]/div[1]/div[1]/label')
+        elem.click()
+
+        elem = driver.find_element(
+          By.CSS_SELECTOR,
+          "button."
+          "app-components-BookFlightForm-bookFlightForm__findFlightBtn--1lbFe")
+        elem.submit()
+
+    # input flight info in subsequent pages
+    else:
+        elem = driver.find_element(
+          By.ID, "airport-autocomplete-.swap0.fromTrip")
+        elem.send_keys(Keys.CONTROL + "a")
+        elem.send_keys(origin)
+
+        elem = driver.find_element(
+          By.XPATH,
+          '//li[@id="airport-autocomplete-.swap0.fromTrip-menu-item-0"]//div')
+        elem.click()
+
+        elem = driver.find_element(By.ID, "airport-autocomplete-.swap0.toTrip")
+        elem.send_keys(Keys.CONTROL + "a")
+        elem.send_keys(dest)
+
+        elem = driver.find_element(
+          By.XPATH,
+          '//li[@id="airport-autocomplete-.swap0.toTrip-menu-item-0"]//div')
+        elem.click()
+
+        elemList = driver.find_elements(
+          By.XPATH, '//div[@class="DateInput DateInput_1"]/input')
+
+        elem = elemList[0]
+        elem.send_keys(Keys.CONTROL + "a")
+        elem.send_keys(depDate)
+
+        elem = elemList[1]
+        elem.send_keys(Keys.CONTROL + "a")
+        elem.send_keys(retDate)
+
+        elem = driver.find_element(
+          By.CSS_SELECTOR,
+          ".atm-c-btn.atm-c-btn--secondary")
+        elem.click()
+
+    time.sleep(15)
+
+    # scrape fare data
+    page_source = driver.page_source
+    soup = BeautifulSoup(page_source, "html.parser")
+    flight_elements = parsePageSource(soup)
+
+    # write to file
+    print(dest + " ********************************************************\n")
+    for i in range(len(flight_elements)):
+        for j in range(len(flight_elements[i])):
+            print(*flight_elements[i][j])
+        print()
+
+    count = 1
```

Comment:
if you are going to keep track of a count, you should at least iterate it so its an actual count (ie `count++`)

Reply:
I actually changed this right after I initiated the pull request.  I was
really just trying to trigger the else statement in the following
iterations.  I realized it was hard to follow, so I changed the counter to
a boolean, mainPage = True
![Screenshot from 2023-01-29 19-57-03](https://user-images.githubusercontent.com/107145275/215371890-5e6c7f20-477b-4a9e-8e9a-45563f8b974b.png)